core/FrontController.php

```diff
@@ -408,9 +408,18 @@ public function init()
 
         // Force the auth to use the token_auth if specified, so that embed dashboard
         // and all other non widgetized controller methods works fine
-        if (Common::getRequestVar('token_auth', false, 'string') !== false) {
+        if (Common::getRequestVar('token_auth', '', 'string') !== ''
+            && Request::shouldReloadAuthUsingTokenAuth(null)) {
             Request::reloadAuthUsingTokenAuth();
+            if (($module !== 'API' || ($action && $action !== 'index')) && Piwik::isUserHasSomeWriteAccess() && !Common::isPhpCliMode()) {
```

Comment:
would it be a good idea to move `($module !== 'API' || ($action && $action !== 'index'))` and ` && !Common::isPhpCliMode()` to another method? something like `isTokenAuthLimitedToWriteAccess()`

Reply:
👍  changed it. will merge if tests pass.